app/assets/javascripts/components/projects/SideBar/index.js

```diff
@@ -1,30 +1,43 @@
 import React from 'react';
 import SideBarButton from './SideBarButton';
-import { reverseColumns } from '../../../actions/projectBoard';
-import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
+import SideBarButtonShape from '../../shapes/sideBarButton';
+import classname from 'classnames';
 
-export const SideBar = ({
-  reverseColumns
+const SideBar = ({
+  buttons
 }) =>
   <div className="SideBar">
     <ul>
-      <SideBarButton
-        description={I18n.t('revert_columns_tooltip')}
-        onClick={reverseColumns}
-        data-id="reverse-button"
-      >
-        <i className="fas fa-redo-alt SideBar__icon"></i>
-      </SideBarButton>
+      {
+        buttons.map(button => {
+            const iconStyle = classname(
+              'SideBar__icon',
+              {
+                'SideBar__icon--is-toggled': button.toggled
+              },
+              button.icon
+            );
+
+            return (
+              <SideBarButton
+                description={button.description}
+                onClick={button.onClick}
+                data-id={button['data-id']}
+                toggled={button.toggled}
```

Comment:
I think you can destructing `button` to pass these props:
```diff
- <SideBarButton
-   description={button.description}
-   onClick={button.onClick}
-   data-id={button['data-id']}
-   toggled={button.toggled}
-   key={button['data-id']}
- >
+ <SideBarButton
+   key={button['data-id']}
+   {...button}
+ >
```

😄

Reply:
Oh, sure. Thanks!